index.js

```diff
@@ -76,6 +99,13 @@ function parse (opts, cb) {
   var pkgPath = opts.path
   var pkg = opts.package
 
+  var extensions = opts.extensions
+  var configuredDetective = opts.detective
```

Comment:
Let's name this `parser` or keep it `detective`. We can also require `detective` when required instead of always:

```js
var detective = opts.detective || 'detective'
var parser = typeof detective === 'string' ? require(detective) : detective

if (typeof detective !== 'function') return cb(new TypeError('Can not load detective'))
```

There's also an inconsistency since `require` will throw here but we're using a callback. May want to `try..catch` that `require`.